src/reformat_libyuv.c

```diff
@@ -77,6 +77,54 @@ avifResult avifImageRGBToYUVLibYUV(avifImage * image, const avifRGBImage * rgb)
     return AVIF_RESULT_NOT_IMPLEMENTED;
 }
 
+// Two-step replacement for AVIF_RGB_FORMAT_RGBA to 8-bit BT.601 full range YUV, which is missing from libyuv.
+static int avifABGRToJ420(const uint8_t * src_abgr,
+                          int src_stride_abgr,
+                          uint8_t * dst_y,
+                          int dst_stride_y,
+                          uint8_t * dst_u,
+                          int dst_stride_u,
+                          uint8_t * dst_v,
+                          int dst_stride_v,
+                          int width,
+                          int height)
+{
+    // A temporary buffer is needed to swap the R and B channels before calling ARGBToJ420().
+    uint8_t * src_argb;
+    const int src_stride_argb = width * 4;
+    const uint64_t soft_allocation_limit = 16384; // Arbitrarily chosen trade-off between CPU and memory footprints.
+    int num_allocated_rows;
+    if ((uint64_t)src_stride_argb * height <= soft_allocation_limit) {
+        // Process the whole buffer in one go.
+        num_allocated_rows = height;
+    } else {
+        // The last row of an odd number of RGB rows to be converted to subsampled YUV is treated differently
+        // by libyuv, so make sure all steps but the last one process an even number of rows.
+        // Try to process as many row pairs as possible in a single step without allocating more than
+        // soft_allocation_limit, unless two rows need more than that.
```

Comment:
Nit: "two rows need more than that", i.e., `(uint64_t)src_stride_argb * 2 > soft_allocation_limit`, is not the only reason we allocate more than `soft_allocation_limit`. We allocate more than `soft_allocation_limit` when the division has a remainder, i.e., when `soft_allocation_limit` is not a multiple of `(uint64_t)src_stride_argb * 2`.

Reply:
Ah yes, my bad. It was not my intent so I changed the formula and removed AVIF_DIV_CEIL().